File: java/27-Remove-Element.java
class Solution {
    public void swap(int[] a, int i, int j) {
        int temp = a[i];
        a[i] = a[j];
        a[j] = temp;
    }
    public int removeElement(int[] nums, int val) {
        int boundary = nums.length-1;
        int i = 0;
        while (i <= boundary) {
            if (nums[i] == val) {
                swap(nums, i, boundary);
                boundary -= 1;
            }
            else {
                i += 1;
            }
        }        
        return i;
    }
}

File: c/27-Remove-Element.c
int removeElement(int* nums, int numsSize, int val){
    int k = 0;
    
    for(int i = 0; i < numsSize; i++)
    {
        if(nums[i] != val)
        {
            nums[k] = nums[i];
            k++;
        }
    }

    return k;
}
